app/services/reporter.rb

```diff
@@ -6,61 +6,80 @@
 # run queries and produce reports from the results, for consumption
 # by preservation catalog maintainers
 class Reporter
-  attr_reader :storage_root, :druids
+  attr_reader :storage_root
 
   # @params [Hash] params used to initialize the Reporter service
   # @return [Reporter] the reporter
   def initialize(params)
     @storage_root = MoabStorageRoot.find_by!(name: params[:storage_root_name])
-    @druids = moab_storage_root_druid_list
+    @msr_names = {}
   end
 
-  # @return [Array] an array of druids on the storage root
-  def moab_storage_root_druid_list
-    druid_array = []
+  def moab_storage_root_list_pres_obj_comp_moabs_relation
     PreservedObject
       .joins(:complete_moabs)
       .where(complete_moabs: { moab_storage_root: storage_root })
-      .select(:druid)
       .order(:druid)
+  end
+
+  # @return [Array] an array of druids on the storage root
+  def druid_csv_list
+    druid_array = [['druid']]
+    moab_storage_root_list_pres_obj_comp_moabs_relation
+      .select(:druid)
       .each_row do |po_hash|
-        druid_array << po_hash['druid']
+        druid_array << [po_hash['druid']]
       end
     druid_array
   end
 
-  # @param [Array] druids - list of druids to output details for
   # @param [Boolean] errors_only (default: false) - optionally only output lines with audit errors
   # @return [Array] an array of hashes with details for each druid provided
-  def moab_detail_for(druids, errors_only: false)
-    detail_array = []
-    druids.each do |druid|
-      preserved_object = PreservedObject.find_by(druid: druid)
-      preserved_object.complete_moabs.each do |cm|
-        next if errors_only && cm.status == 'ok'
-        detail_array << { druid: druid,
-                          status: cm.status,
-                          status_details: cm.status_details,
-                          last_moab_validation: cm.last_moab_validation,
-                          last_checksum_validation: cm.last_checksum_validation,
-                          storage_root: cm.moab_storage_root.name,
-                          from_storage_root: cm.from_moab_storage_root&.name }
+  def moab_detail_csv_list(errors_only: false)
```

Comment:
i'm not a huge fan of named params (params hash - whatever it's called) when there are a small number and the method is unlikely to be used outside the single coded caller ...

Reply:
@ndushay :laughing: welllllll, I'm a huge fan of preferring keyword arguments over standard arguments for two reasons:

1. They are not ordered so a dependency is removed: needing to know what order the args are passed in. We still do have the dependency of needing to know what args are required, but that's true of both styles);
1. They are more intention-revealing than standard arguments. How many times have you seen code like `do_the_thing(42, true)` and wondered WTH `42` and `true` mean? So then you have to go trawling in another codebase, potentially, for enlightenment. Compare with `do_the_thing(max_records: 42, skip_accessioned: true)`. This also means we need to be vigilant about coming up with good names, avoiding names like `process` and `data` and such, but that (again) is true in both styles.

Reply:
i also prefer keyword args for the reasons @mjgiarlo gives.